migration/register.go

```diff
@@ -0,0 +1,120 @@
+package migration
+
+import (
+	"reflect"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+// Payload is implemented by both weave.Msg and models used by the orm package.
+// Schema migration supports both of those data types.
+type Payload interface {
+	GetHeader() *weave.Header
```

Comment:
We may need a new word. I had to read this twice, thinking it was the tendermint block header.

Maybe `Metadata`? `VersionInfo`?

Reply:
Also, just as general go best practice - we can omit Get and just use `Header|Metadata..`

Reply:
This comes from protobuf definitions. They define public members eg. `Header` and public accessors `GetHeader()`. The accessors are mainly there to fulfill interfaces.

Given this is designed around protobuf messages, naming makes sense

Reply:
This is generated by the protobuf compiler :man_shrugging:

Reply:
I'm pretty crap at naming but if you like Metadata as well, then let's go for it.
Nice and generic, and a superset of schema/version info

Reply:
I will go with `Metadata` as it is better than `Heade`. We can always change it later :see_no_evil: